frontend/www/js/omegaup/arena/arena.ts

```diff
@@ -1894,10 +1895,59 @@ export class Arena {
     }
     $('#overlay form:not([data-run-submit])').hide();
     $('#overlay').show();
-    api.Run.details({ run_alias: showRunMatch[1] })
+    const guid = showRunMatch[1];
+    api.Run.details({ run_alias: guid })
       .then(time.remoteTimeAdapter)
       .then((data) => {
-        this.displayRunDetails(showRunMatch[1], data);
+        if (
+          data.show_diff === 'none' ||
+          (this.options.contestAlias && this.options.contestAlias !== 'admin')
+        ) {
+          this.displayRunDetails(guid, data);
+          return;
+        }
+        fetch(`/api/run/download/run_alias/${guid}/show_diff/true/`)
+          .then((response) => {
+            if (response.status === 200 || response.status === 0) {
+              return Promise.resolve(response.blob());
+            } else {
+              return Promise.reject(new Error(response.statusText));
+            }
```

Comment:
acá se puede usar el guard clause pattern:

```suggestion
            if (!response.ok) {
              return Promise.reject(new Error(response.statusText));
            }
            return Promise.resolve(response.blob());
```

Reply:
Listo